qcodes/instrument/base.py

```diff
@@ -114,24 +118,31 @@ def add_parameter(self, name: str,
                 unit of the new parameter is inconsistent with the existing
                 one.
         """
-        existing_parameter = self.parameters.get(name, None)
-
-        if existing_parameter:
+        if "bind_to_instrument" not in kwargs.keys():
```

Comment:
``in`` will look in keys already ;)
```suggestion
        if "bind_to_instrument" not in kwargs:
```

Reply:
Imho being explicit makes it more readable but I don't care strongly